File: 134-GasStation/134-GasStation.py
# Last updated: 3/31/2026, 9:39:47 PM
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        if (sum(gas)<sum(cost)):
            return -1
        fuel = 0
        ind = 0
        for i in range(len(gas)):
            f = gas[i] - cost[i]
            fuel+= f
            if fuel<0:
                ind = i+1
                fuel = 0
        return ind

File: 134-GasStation/134-GasStation.cpp
// Last updated: 3/31/2026, 9:39:42 PM
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int expend = 0;
        int fuel = 0;
        int ind = 0;
        for(int i =0; i< gas.size();i++){
            fuel += (gas[i]-cost[i]);
            if (fuel<0){
                expend += fuel;
                fuel = 0;
                ind = i+1;
            }
        }
        return (fuel+expend)>=0? ind:-1;
    }
};